### PR TITLE
Notify users day before retention will delete a file

### PR DESCRIPTION
**Pull request by rullzer**
TODO:
- [x] Checkbox in admin settings
- [x] Display actual notification

### CODE REVIEW

**Comment by rullzer** [comment]
admin

notification

File: lib/BackgroundJob/RetentionJob.php [hunk 1]
```diff
@@ -23,19 +23,23 @@
...
 use OCP\Files\Node;
 use OCP\Files\NotFoundException;
 use OCP\Files\NotPermittedException;
 use OCP\Files\IRootFolder;
 use OCP\ILogger;
+use OCP\Notification\IManager as NotificationManager;
 use OCP\SystemTag\ISystemTagManager;
 use OCP\SystemTag\ISystemTagObjectMapper;
 use OCP\SystemTag\TagNotFoundException;
+use function DeepCopy\deep_copy;
```
[the remark this answers is not in the document]
Comment (rullzer) [reply lib/BackgroundJob/RetentionJob.php]: Yes will fix that

**Comment by juliushaertl** [comment]
@rullzer Do we need to backport this?

**Comment by rullzer** [comment]
> @rullzer Do we need to backport this?

yes

**Pull request merged**